scripts/README.md

```diff
@@ -60,3 +61,61 @@ kubectl get pods || fail_test
 
 success
 ```
+
+## Using the `release.sh` helper script
+
+This is a helper script for Knative release scripts. To use it:
+
+1. Source the script.
+
+1. Call the `parse_flags()` function passing `$@` (without quotes).
+
+1. Call the `run_validation_tests()` function passing the script or executable that
+runs the release validation tests. It will call the script to run the tests unless
+`--skip_tests` was passed.
+
+1. Write logic for the release process. Call `publish_yaml()` to publish the manifest(s),
+`tag_releases_in_yaml()` to tag the generated images, `branch_release()` to branch
```

Comment:
once the environment variable is set, why the caller still need to know which release function to call? 
Wouldn't it be better if there is a generic function say do_release() which will determine whether to call publish or tag or branch based on environment variable?

Reply:
The idea is that this script is a helper, not a framework. I don't want to make release scripts rigid, that will just make the helper more complex and incorporate too specific bits. If you check the release scripts, you'll see that they have too different approaches, which is not really suitable for what you're proposing (I mean, not without the disadvantages I mentioned).